worker/worker.go

```diff
@@ -959,14 +962,15 @@ func (w *worker) accountsHandlerGET(jc jape.Context) {
 }
 
 // New returns an HTTP handler that serves the worker API.
-func New(masterKey [32]byte, id string, b Bus, sessionReconectTimeout, sessionTTL, interactionsFlushInterval, uploadSectorTimeout time.Duration) (http.Handler, func() error, error) {
+func New(masterKey [32]byte, id string, b Bus, sessionReconectTimeout, sessionTTL, interactionsFlushInterval, uploadSectorTimeout time.Duration, l *zap.Logger) (http.Handler, func() error, error) {
 	w := &worker{
 		id:                        id,
 		bus:                       b,
 		pool:                      newSessionPool(sessionReconectTimeout, sessionTTL),
 		masterKey:                 masterKey,
 		interactionsFlushInterval: interactionsFlushInterval,
 		uploadSectorTimeout:       uploadSectorTimeout,
+		logger:                    l.Sugar().Named(id),
```

Comment:
I originally had `fmt.Sprintf("worker_%s", id)`, our default worker id is "worker" so that made me use the `id` because I figured people could just use worker_foo as worker id. I guess. `Named().Named()` is better I never considered that.